.github/workflows/ci.yaml

```
@@ -0,0 +1,388 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - 2.*
+  pull_request: ~
+
+env:
+  CACHE_VERSION: 1
+  DEFAULT_PYTHON: 3.6
+  PRE_COMMIT_CACHE: ~/.cache/pre-commit
+
+jobs:
+  prepare-base:
+    name: Prepare base dependencies
+    runs-on: ubuntu-latest
+    outputs:
+      python-key: ${{ steps.generate-python-key.outputs.key }}
+      pre-commit-key: ${{ steps.generate-pre-commit-key.outputs.key }}
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v2.3.4
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        id: python
+        uses: actions/setup-python@v2.2.1
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+      - name: Generate partial Python venv restore key
+        id: generate-python-key
+        run: >-
+          echo "::set-output name=key::base-venv-${{ env.CACHE_VERSION }}-${{
+            hashFiles('setup.cfg', 'requirements_test.txt', 'requirements_test_min.txt')
```

```suggestion
            hashFiles('setup.cfg', 'requirements_test.txt', 'requirements_test_pre_commit.txt',
                      'requirements_test_min.txt', 'requirements_test_brain.txt')
```
Besides `requirements_test_brain.txt`, we probably need to add `requirements_test_pre_commit.txt` as well -> for `pylint`.